backend/spec/requests/roles_request_spec.rb

```diff
@@ -2,14 +2,40 @@
 
 RSpec.describe 'RolesController', type: :request do
   describe '#list' do
-    let!(:role) { create(:role) }
+    let!(:role_moderator) { create(:role_moderator) }
+    let!(:role_admin) { create(:role_admin) }
+    let!(:role_teacher) { create(:role_teacher) }
+    let!(:role_student) { create(:role_student) }
+    let(:user_admin) { create(:user) }
 
-    before { get api_v1_roles_path }
+    describe 'List all roles when admin' do
+      before do
+        get api_v1_roles_path, headers: auth_headers(user: user_admin)
+      end
 
-    it { expect(response).to have_http_status :ok }
+      it { expect(response).to have_http_status :ok }
 
-    it 'matches surveys attributes' do
-      expect(response_body).to match([{ 'id' => role.id, 'role_type' => role.role_type }])
+      it 'matches surveys attributes' do
+        expect(response_body).to match([{ 'id' => role_moderator.id, 'role_type' => role_moderator.role_type }, { 'id' => role_admin.id, 'role_type' => role_admin.role_type },
```

Comment:
maybe you should create a let `not_admin_roles` and `all_roles`. It will make this and the other spec more legible

Reply:
Done!